mlflow/store/model_registry/dbmodels/models.py

```diff
@@ -84,4 +86,67 @@ def to_mlflow_entity(self):
         return ModelVersion(self.name, self.version,
                             self.creation_time, self.last_updated_time, self.description,
                             self.user_id, self.current_stage, self.source, self.run_id,
-                            self.status, self.status_message)
+                            self.status, self.status_message,
+                            [tag.to_mlflow_entity() for tag in self.model_version_tags])
+
+
+class SqlRegisteredModelTag(Base):
+    __tablename__ = 'registered_model_tags'
+
+    name = Column(String(256), ForeignKey('registered_models.name',
+                                          onupdate='cascade',
+                                          ondelete='cascade'))
+
+    key = Column(String(250), nullable=False)
+
+    value = Column(String(5000), nullable=True)
+
+    # linked entities
+    registered_model = relationship('SqlRegisteredModel',
+                                    backref=backref('registered_model_tags',
+                                                    cascade='all'))
+
+    __table_args__ = (
+        PrimaryKeyConstraint('key', 'name', name='registered_model_tag_pk'),
+    )
+
+    def __repr__(self):
+        return '<SqlRegisteredModelTag ({}, {}, {})>'.format(self.name, self.key, self.value)
+
+    # entity mappers
+    def to_mlflow_entity(self):
+        return RegisteredModelTag(self.key, self.value)
+
+
+class SqlModelVersionTag(Base):
+    __tablename__ = 'model_version_tags'
+
+    name = Column(String(256))
+
+    version = Column(Integer)
+
+    key = Column(String(250), nullable=False)
+
+    value = Column(String(5000), nullable=True)
+
+    # linked entities
+    model_version = relationship('SqlModelVersion', foreign_keys=[name, version],
+                                 backref=backref('model_version_tags',
+                                                 cascade='all'))
+
+    __table_args__ = (
+        PrimaryKeyConstraint('key', 'name', 'version', name='model_version_tag_pk'),
+        ForeignKeyConstraint(
+            ('name', 'version'),
+            ('model_versions.name', 'model_versions.version'),
+            onupdate='cascade',
+            ondelete='cascade')
```

Comment:
let's add tests to confirm `ondelete` and `onupdate` updates in this table correctly. We should also test this manually on at least SQLite and MySQL

Reply:
This will be tested in the next PR at the store level implementations